src/System.Text.JsonLab/System/Text/Json/JsonReader.cs

```diff
@@ -31,7 +52,7 @@ namespace System.Text.JsonLab
 
         // These properties are helpers for determining the current state of the reader
         internal bool InArray => !InObject;
-        internal bool InObject => (_containerMask & 1) != 0;
+        internal bool InObject => Depth <= StackFreeMaxDepth ? (_containerMask & 1) != 0 : _stack.Peek() != 0;
```

Comment:
Would it make sense to compute it once inside Start/Stop and then cache the result in bool _inObject field?

Reply:
Fixed in https://github.com/dotnet/corefxlab/pull/2484/commits/ba0d2b39c057731bc39ddb62e2364319887aa8ad (https://github.com/dotnet/corefxlab/pull/2484).